packages/ui/src/views/secrets/secret-reference/secret-reference.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+
+import { Button, ButtonGroup, Icon, StackedList } from '@/components'
+import { EntityReference, EntityRendererProps, ScopeSelectorProps } from '@/views/entity-reference'
+
+import { DirectoryType, SecretItem } from '../types'
+
+export interface SecretReferenceProps {
+  // Data
+  secretsData: SecretItem[]
+  childFolder: string | null
+  parentFolder: string | null
+
+  // State
+  selectedEntity: SecretItem | null
+
+  // Callbacks
+  onSelectEntity: (entity: SecretItem) => void
+  onScopeChange: (direction: DirectoryType) => void
+  onCancel?: () => void
+  isLoading?: boolean
+}
+
+// Component for selecting existing secrets
+export const SecretReference: React.FC<SecretReferenceProps> = ({
+  // Data
+  secretsData,
+  childFolder,
+  parentFolder,
+
+  // State
+  selectedEntity,
+
+  // Callbacks
+  onSelectEntity,
+  onScopeChange,
+  onCancel,
+  isLoading
+}) => {
+  // Define available scopes
+
+  // Custom entity renderer for secrets
+  const renderEntity = (props: EntityRendererProps<SecretItem>) => {
+    const { entity, isSelected, onSelect } = props
+
+    return (
+      <StackedList.Item
+        onClick={() => onSelect(entity)}
+        className={isSelected ? 'bg-background-4' : ''}
+        thumbnail={<Icon name="secrets" size={16} className="text-foreground-5" />}
+        actions={
+          <Button
+            variant="default"
+            size="sm"
+            onClick={e => {
+              e.stopPropagation()
+              onSelect(entity)
+            }}
+          >
+            Select
+          </Button>
+        }
+      >
+        <StackedList.Field title={entity.secret.name} description={entity.secret.description || undefined} />
+      </StackedList.Item>
+    )
+  }
+
+  // Custom scope selector renderer
+  const renderScopeSelector = (props: ScopeSelectorProps<string>) => {
```

Comment:
i'd like to avoid overriding the scope selector. the default renderer should cover the standard usecase, since it'll largely be the same for all entities